\section*{Methods and process}
In order to measure the CNV of the genes on the Chimpanzee X chromosome, we aligned the reference chromosome X (Pan\_Tro\_3) to itself and created dotplots in partially overlapping windows of 500 Kbp. These dotplots depict internal duplicates inside and immediately surrounding the sequence in the window. By manually browsing the catalog of 500 overlapping windows, we decided manually for each of the ~1000 genes in the chromosome if they showed enough internal duplicates to be included in the subsequent analysis. By concatenating these selected genes into an artificial chromosome (AC), and then mapping the reads from each individual to this AC, we were able to measure the relative CNV of the genes. Because the dotplot method takes only adjacent duplicates into account (limited window size), this method makes it possible to identify only the duplicated regions that reside inside and near by the focal gene. As a solution to this problem, we decided instead to compute the CNV of all annotated genes in the chromosome. We did this by mapping the reads from each individual to a well annotated reference chromosome (Pan\_tro\_3). To get a relative measure of CNV for each gene we then measured the median coverage of each gene here on. This method might be more computationally intensive, but it has the advantage of eliminating selection bias, as well as the advantage, that reads that map better outside the selected genes, will not affect the coverage of the genes. We filtered to keep only primary mapped reads with a mapping quality of at least 50, a consecutive mapping of at least 100 bp and an overall maximum nucleotide mismatch of at most 2. Because this project uses publicly available data from different projects, the quality and overall coverage might differ (see table??). As the median coverage of the complete X chromosome differs 10-fold ??, we needed a way to convert relative CNV to absolute CNV. The dystrophin (DMD ??ENSPTRG{0}44) gene is very long (?? Kbp), single copy and has been used in similar studies [Lucotte 2019] for the same purpose. By dividing the coverage of each gene with the coverage of DMD in the same individual, we obtained an absolute CN measure for each gene on the X chromosome, in each individual. 

We define gene copy number variation median of copy numbers in all individuals. This relates to the variance of the medians, as median and standard deviation is highly correlated. 
