\documentclass[fleqn,12pt]{wlscirep}
\usepackage[utf8]{inputenc}
\usepackage[T1]{fontenc}

% Carls packages
\usepackage{wrapfig}
%\usepackage{pst-pdgr}
%\usepackage[all]{genealogytree}
\title{Copy Number Variation in The Chimpanzee X Chromosome}

\author{Project in Bioinformatics, 5 ECTS, Fall 2019 - Carl Mathias Kobel}


\setlength{\parskip}{0.4em}
\setlength{\parindent}{0pt}



\begin{abstract}
Counting the copy number of genes in an individual can be done by read mapping to a reference, and measuring the coverage of each gene. Normalizing this measure, such that it can be compared across individuals can be done either by scaling with the coverage of a long single-copy gene (i.e. DMD) or by assuming that the mean of the copy number distribution, for all linked genes, is equal to one. In this study, both methods are discussed, and the latter method is used for a search of genes, on the Chimpanzee X chromosome, with high copy number variation.
It is observed that a number of previously undescribed genes with high copy number show signs of testis-expression. These genes might be part of a meiotic drive process. Some of these genes also show a strong sex-grouping which might be due to the failure to include the X-degenerate region of the Y chromosome in the mapping-reference.
Anyhow, because expression data on Chimpanzee genes is in many occasions not readily available, it is not possible to immediately conclude that any of these genes are indeed related to meiotic drive.
\end{abstract}

\begin{document}

\flushbottom
\maketitle


\thispagestyle{empty}


\input{1_Introduction.tex}


\input{2_Methods.tex}


\input{3_Results.tex}


\input{4_Conclusion.tex}

\bibliography{sample}

\input{5_Appendix.tex}








\end{document}